smartsim/_core/control/manifest.py

```diff
@@ -109,6 +109,12 @@ def all_entity_lists(self) -> t.List[EntitySequence[SmartSimEntity]]:
 
         return _all_entity_lists
 
+    @property
```

Comment:
The new property `all_entities` is only used to evaluate as a boolean for presence of entities in the manifest. If we didn't already have a ton of properties listing each individual item, I probably wouldn't notice it. 

### Do you think the API is better if we instead indicate what you want to know directly? 

```py
@property
def has_deployable(self) -> bool:
   """Return True if the manifest contains entities that must be physically deployed"""
   return bool(self._deployables)

```
which results in
```html
{%- if manifest.has_deployables %}
...
```

### Could we just avoid adding another list property? The current usage is only to show models, so we could just do:

```html
{%- if manifest.models %}
...
```

Reply:
### Creating a View Model

An alternative is to build a facade over the manifest that gives the preview-centric view properties. This is a "ViewModel" whose purpose is to aggregate or reshape existing data into a format that is useful for display, but doesn't fulfill the original purpose of the object you're interested in.

Views don't make a lot of immediate sense, but by the time it does we might have added a ton of manifest properties that we no longer recognize as only used by the manifest. 

Consider:

```py

class ManifestPreviewView:
  def __init__(self, manifest: Manifest) -> None:
    self._manifest = manifest

  def has_deployables(self) -> bool:
    return any(manifest.models)
  
  def __attr__(self, attr_name: string) -> t.Any:
    if hasattr(self._manifest, attr_name):
      return getattr(self._manifest, attr_name)

```
 
This creates a purpose built way to aggregate things used in a preview and doesn't start to degrade the API of an existing object